### PR TITLE
DEPR: downcast inferring round floats to ints

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

xref #40988

### CODE REVIEW

File: pandas/conftest.py [hunk 1]
```diff
@@ -145,6 +145,10 @@ def pytest_collection_modifyitems(items, config) -> None:
             "(Series|DataFrame).bool is now deprecated and will be removed "
             "in future version of pandas",
         ),
+        (
+            "pandas.core.generic.NDFrame.clip",
```
Comment (phofl) [inline pandas/conftest.py:149]: Should we add this one to the whatsnew as well?

File: doc/source/whatsnew/v2.1.0.rst [hunk 2]
```diff
@@ -261,6 +261,8 @@ Deprecations
 - Deprecated logical operation between two non boolean :class:`Series` with different indexes always coercing the result to bool dtype. In a future version, this will maintain the return type of the inputs. (:issue:`52500`, :issue:`52538`)
 - Deprecated allowing ``downcast`` keyword other than ``None``, ``False``, "infer", or a dict with these as values in :meth:`Series.fillna`, :meth:`DataFrame.fillna` (:issue:`40988`)
 - Deprecated constructing :class:`SparseArray` from scalar data, pass a sequence instead (:issue:`53039`)
+- Deprecated downcasting behavior in :meth:`Series.interpolate`, :meth:`Series.fillna`, :meth:`DataFrame.interpolate`, :meth:`DataFrame.fillna`, with downcast="infer" and floating dtypes; in a future version these will not cast all-round floats to integer dtype, explicitly cast the result instead (:issue:`40988`)
```
Comment (phofl) [inline doc/source/whatsnew/v2.1.0.rst:264]: What's the keyword doing now? Only casting from int64 -> int8?

Comment (jbrockmendel) [reply doc/source/whatsnew/v2.1.0.rst:264]: Just object dtype

File: pandas/conftest.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/conftest.py:149]: will update

Comment (jbrockmendel) [reply pandas/conftest.py:149]: updated + green

**Comment by jbrockmendel** [comment]
I guess we can do this and then follow up by deprecating the keyword entirely?  its a bit roundabout, but gives anyone who objects lots of chances to do so

**Comment by jbrockmendel** [comment]
@phofl gentle ping

**Comment by jbrockmendel** [comment]
Closing in favor of #53656

**Pull request closed without merging**